src/memory.rs

```diff
@@ -279,7 +285,7 @@ impl<'a, 'tcx> Memory<'a, 'tcx> {
             Some(alloc) => Ok(alloc),
             None => match self.functions.get(&id) {
                 Some(_) => Err(EvalError::DerefFunctionPointer),
-                None if id == ZST_ALLOC_ID => Err(EvalError::ZstAllocAccess),
+                None if id == ZST_ALLOC_ID => Err(EvalError::InvalidMemoryAccess),
```

Comment:
I'm thinking of renaming `ZST_ALLOC_ID` to `UNALLOCATED_ID` or something like that, since it's now used for all pointers that have no real backing allocation.